openmcyclus/depletion.py

```diff
@@ -62,6 +61,56 @@ def read_model(self):
         model = openmc.model.model.Model.from_xml(**model_kwargs)
         return model
 
+    def update_materials(self, comp_list):
+        '''
+        Read in the material compositions of the fuel assemblies present
+        in the reactor to be transmuted. Then modify the composition of
+        the pre-defined materials to match the compositions from
+        Cyclus.
+
+
+        Parameters:
+        -----------
+        comp_list: list of dicts
+            list of the fresh fuel compositions present in the core
+            at the calling of the transmute function.
+
+        Returns:
+        --------
+        material_ids: list of strs
+            material id numbers for the OpenMC model
+
+        Outputs:
+        --------
+        materials.xml: file
+            updated XML for OpenMC with new compositions
+
+        '''
+        openmc_materials = ET.parse(str(self.path / "materials.xml"))
+        openmc_root = openmc_materials.getroot()
+        material_ids = []
+
+        for child in openmc_root:
+            if '_' in child.attrib['name']:
+                underscore_index = child.attrib['name'].index('_')
+                assembly_number = child.attrib['name'][underscore_index + 1:]
+                material_ids.append(child.attrib['id'])
+                if child.attrib['name'][:underscore_index] == 'assembly':
+                    for material in child.findall('nuclide'):
+                        child.remove(material)
+                    new_comp = comp_list[int(assembly_number) - 1]
+                    for nuclide in new_comp:
```

Comment:
Doubling down on my previous suggestions -- `for nuclide in new_comp` is better than `for i in range(len(new_comp)): new_comp[i]` so keep doing this!